### PR TITLE
Build the distribution package in Gradle

### PR DESCRIPTION
**Pull request by emanuelez**
This commit does the following:
 * converts the build-distribution script to Gradle
 * automates the Realm version of distribution examples
 * generates the distribution archive

### CODE REVIEW

**Comment by emanuelez** [comment]
@bmunkholm @cmelchior @kneth

File: realm/build.gradle [hunk 1]
```diff
@@ -138,3 +138,84 @@ tasks.preBuild {
     dependsOn compileAnnotations
     dependsOn buildApt
 }
+
+def examples = [
```
Comment (cmelchior) [inline realm/build.gradle]: Do we really want to enumerate all the projects we want to include. Wouldn't it be better to name the ones we *don't* want. I would assume that would be easier to maintain as the most common use case would be to include them?

**Comment by cmelchior** [comment]
Apart from comments :+1:

File: realm/build.gradle (hunk 1 above)
Comment (emanuelez) [inline realm/build.gradle]: My evil master-plan is to completely automate the generation of the distribution folder.
But right now it's nice to have the flexibility to call the example folders in any way, without having to follow a convention.

Comment (cmelchior) [inline realm/build.gradle]: :+1:

**Pull request merged**